src/components/ShareLocationButton.js

```diff
@@ -0,0 +1,34 @@
+import React, { useContext } from 'react';
+
+import { MapCenterContext } from '../context/MapCenterContext';
+
+const ShareLocationButton = () => {
+  const valueCenterMap = useContext(MapCenterContext);
+  const { setNewCenterMap } = valueCenterMap;
+
+  let id;
+  const getLocation = (e) => {
+    id = navigator.geolocation.watchPosition(success, error);
+  };
+
+  function success(position) {
+    const lat = position.coords.latitude;
+    const lng = position.coords.longitude;
+    const newMapCenter = { lat, lng };
+    setNewCenterMap(newMapCenter);
+  }
+
+  function error() {
+    const defaultCenterMap = {
+      lat: 48.8566,
+      lng: 2.3522,
+    };
+    setNewCenterMap(defaultCenterMap);
+    navigator.geolocation.getCurrentPosition(success, error);
```

Comment:
Help me understand the purpose of this line? We got an error, but are we trying again one more time?

Reply:
@segdeha : It's related to this subtask "If the user does not accept, we leave the map in whatever is its last good state".

I'm trying to find out the 'last' position shared before we are going to stop 'sharing location'. I think it could cause some confusion because of the work related to sharing our position in our app. It's a little incomplete because this week we worked about 'start' sharing our position, but not 'stop'.

I believe I forget to check if `navigator.geolocation !== null`. Thank you!

There are two moments when this code snippet is called:
1. Start the app, click the 'Share location' button, browser prompt is open, and you decided to 'block' it.
2. Start the app, click the 'Share location' button, you change your current position (maybe you leave your house), and you stop to share your location, but you need to 'save' this last position. (I think we will complete our work next week because there is another [Jira ticket - TCL30-14](https://the-collab-lab.atlassian.net/browse/TCL30-14) about stop sharing location, but **It's possible I'm wrong**, and I can try other ideas).

I hope I have understood this subtask. Of course, I hope I could have shared my reasons for this line `navigator.geolocation.getCurrentPosition`.